sdk/core/azure-core/azure/core/pipeline/transport/__init__.py

```diff
@@ -24,137 +24,92 @@
-                    except ImportError:
-                        raise ImportError("aiohttp package is not installed")
-                return name
+        
+def __dir__():
+    return __all__
 
-        else:
-            try:
-                from ._aiohttp import AioHttpTransport, AioHttpTransportResponse
-                __all__.extend([
-                    'AioHttpTransport',
-                    'AioHttpTransportResponse',
-                ])
-            except ImportError:
-                pass  # Aiohttp not installed
-    except (ImportError, SyntaxError):
-        pass  # Asynchronous pipelines not supported.
+def __getattr__(name):
+    if name == 'AsyncioRequestsTransport':
+        try:
+            from ._requests_asyncio import AsyncioRequestsTransport
+            return AsyncioRequestsTransport
+        except ImportError:
+            pass
+    if name == 'AsyncioRequestsTransportResponse':
+        try:
+            from ._requests_asyncio import AsyncioRequestsTransportResponse
+            return AsyncioRequestsTransportResponse
+        except ImportError:
+            pass
+    if name == 'AsyncHttpTransport':
+        try:
+            from ._base_async import AsyncHttpTransport
+            return AsyncHttpTransport
+        except ImportError:
+            pass
+    if name == 'AsyncHttpResponse':
+        try:
+            from ._base_async import AsyncHttpResponse
+            return AsyncHttpResponse
+        except ImportError:
+            pass
+    if name == 'RequestsTransport':
+        try:
+            from ._requests_basic import RequestsTransport
+            return RequestsTransport
+        except ImportError:
+            raise ImportError("requests package is not installed")
+    if name == 'RequestsTransportResponse':
+        try:
+            from ._requests_basic import RequestsTransportResponse
+            return RequestsTransportResponse
+        except ImportError:
+            raise ImportError("requests package is not installed")
+    if name == 'AioHttpTransport':
+        try:
+            from ._aiohttp import AioHttpTransport
+            return AioHttpTransport
+        except ImportError:
+            raise ImportError("aiohttp package is not installed")
+    if name == 'AioHttpTransportResponse':
+        try:
+            from ._aiohttp import AioHttpTransportResponse
+            return AioHttpTransportResponse
+        except ImportError:
+            raise ImportError("aiohttp package is not installed")
+    if name == 'TrioRequestsTransport':
+        try:
+            from ._requests_trio import TrioRequestsTransport
+            return TrioRequestsTransport
+        except ImportError:
+            raise ImportError("trio package is not installed")
+    if name == 'TrioRequestsTransportResponse':
+        try:
+            from ._requests_trio import TrioRequestsTransportResponse
+            return TrioRequestsTransportResponse
```

Comment:
Same as above:
This is interesting, as it may fail because trio is not installed, or because requests is not installed. Maybe sniff the ImportError?

Reply:
Looking into the error message was the best way. Thinking whether to mod the error message in case requests is missing, that its needed for trio transport. Maybe could be confusing to get an error simply about requests while asking for trio.

Reply:
I like what you did above:
```
        except ImportError as ex:
            if ex.msg.endswith("'requests'"):
                raise ImportError("requests package is not installed")
            else:
                raise ImportError("trio package is not installed")
```

Customer will get the full stacktrace anyway, and one of those should say they tried to import `TrioRequestsTransportResponse`, which should be enough to connect the dots.